src/microbe_stage/BioProcess.cs

```diff
@@ -64,6 +64,25 @@ public void Check(string name)
         TranslationHelper.CopyTranslateTemplatesToTranslateSource(this);
     }
 
+    public float GetRateWithConditions(BiomeConditions conditions)
+    {
+        var rate = 1.0f;
+
+        foreach (var input in Inputs)
+        {
+            if (input.Key.IsEnvironmental)
+            {
+                foreach (var compound in conditions.AverageCompounds)
+                {
+                    if (compound.Key.Equals(input.Key))
+                        rate = compound.Value.Ambient / rate;
```

Comment:
I think that the problem with NaN compounds comes from here. To get NaN, you usually have to divide by zero, which seems possible here. If `compound.Value.Ambient` = 0, then on the next iteration, a number will be divided by rate, which is 0. Maybe there was supposed to be multiplication here, instead of division?